### DISCUSSION TITLE
Conditional Dynamic CSS Import Issue in Next.js

### DISCUSSION THREAD

**Discussion by AsadRazaJutt1** [vercel/next.js#69994] [Help]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/sparkling-water-g3ns6c?file=%2Fapp%2Flayout.tsx%3A2%2C19

### To Reproduce

1. Click on [Codesandbox](https://codesandbox.io/p/devbox/sparkling-water-g3ns6c?file=%2Fapp%2Flayout.tsx%3A2%2C19).
2. Go to layout.tsx.
When I load CSS using import "./app.css", it works fine. However, if I try to load it dynamically using const css = "./app.css" and then import(css), it doesn't work.

### Current vs. Expected behavior

CSS is not loading when I try like this  css = "./app.css" and then import(css).
but when I try like this `import "./app.css"` style work properly.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home
  Available memory (MB): 12200
  Available CPU cores: 4
Binaries:
  Node: 18.20.4
  npm: N/A
  Yarn: N/A
  pnpm: 8.15.4
Relevant Packages:
  next: 14.2.5 // There is a newer version (14.2.6) available, upgrade recommended!
  eslint-config-next: 14.2.5
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.5.4
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

create-next-app

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

I tested on 15.0.0-canary.130

**Comment by samcx** [comment]
@AsadRazaJutt1 Is it possible to explain why you wish to import CSS like this? I would think you should just dynamically import a Component that has it's own CSS Modules, instead of trying to dynamically import CSS itself.

**Comment by AsadRazaJutt1** [comment]
@samcx we are [Dukan](https://www.dukan.pk/) migrating from react to NextJs. we have thousands of stores each store may have their own CSS file. So, we need to import CSS dynamically. 

For now, we have achieved this by 

![code](https://github.com/user-attachments/assets/68498c3f-1c64-44a2-8488-592dc122d0ac)

**Comment by samcx** [comment]
Thank you for the additional context.

```
const css = "./app.css"
import(css)
```

I don't think this is possible because the CSS files aren't supposed to exist together like that in Production.

Are all your CSS files all in `/public`? And how are you getting `initialData`?

Maybe you can use the `<Script>` Component to dynamically load a style `<link>` of dangerously setting innerHTML → https://v0.dev/chat/B1rQW0jtzuy.

Since this isn't really a bug currently with Next.js, I'll be converting this to a discussion.
